zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/approval/ApproveTask.java

```diff
@@ -329,18 +331,72 @@ private static ContentDetailResolver getDefaultContentDetailResolver() {
 
     private void validate() {
         if (collection == null) {
-            throw new IllegalArgumentException("approval task unsuccesful: collection required but was null");
+            throw new IllegalArgumentException("approval task unsuccessful: collection required but was null");
         }
         if (collection.getDescription() == null) {
-            throw new IllegalArgumentException("approval task unsuccesful: collection.description required but was " +
+            throw new IllegalArgumentException("approval task unsuccessful: collection.description required but was " +
                     "null");
         }
         if (session == null) {
-            throw new IllegalArgumentException("approval task unsuccesful: as session required but was null");
+            throw new IllegalArgumentException("approval task unsuccessful: as session required but was null");
         }
         if (StringUtils.isEmpty(session.getEmail())) {
-            throw new IllegalArgumentException("approval task unsuccesful: as session.email required but was null/empty");
+            throw new IllegalArgumentException(
+                    "approval task unsuccessful: as session.email required but was null/empty");
         }
-        info().data("collectionId", collection.getDescription().getId()).log("approval task: validation sucessful");
+        info().data("collectionId", collection.getDescription().getId()).log("approval task: validation successful");
+    }
+
+    protected void uploadNewEndpoint(Collection collection, CollectionReader collectionReader) throws ZebedeeException, IOException {
+        uploadWhitelistedFiles(collection, collectionReader);
+    }
+    
+    protected void uploadWhitelistedFiles(Collection collection, CollectionReader collectionReader) throws ZebedeeException, IOException {
+        for (String uri : collectionReader.getReviewed().listUris()) {
+            if (uri.endsWith(".csv") || uri.endsWith(".xlsx") || uri.endsWith(".xls") || uri.endsWith(".csdb")) {
+                String fileName = uri.substring(1);
+                Resource myFile = collectionReader.getResource(fileName);
+                if (DatasetWhitelistChecker.isWhitelisted(myFile.getName())) {
+                    info().log("File is whitelisted");
+                    uploadFile(myFile, fileName, collection.getDescription().getId());
+                } else {
+                    info().log("File is not whitelisted");
+                }
+            }
+        }
+    }
+    
+    protected void uploadFile(Resource myFile, String fileName, String collectionId) throws ZebedeeException, IOException {
+        File file = new File("afile");
+        try (FileOutputStream outputStream = new FileOutputStream(file)) {
+            IOUtils.copy(myFile.getData(), outputStream);
+        } catch (FileNotFoundException e) {
+            info().log("file not found");
```

Comment:
This log information should be set as an error - please see https://github.com/ONSdigital/zebedee/blob/develop/zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/approval/ApproveTask.java#L201 as an example